tests/server/routes/users-test.js

```diff
@@ -0,0 +1,218 @@
+// Instantiate all models
+var expect = require('chai').expect;
+
+var Sequelize = require('sequelize');
+
+var db = require('../../../server/db');
+
+var supertest = require('supertest');
+
+describe('Users Route', function () {
+
+    var app, User, Order, user;
+
+    beforeEach('Sync DB', function () {
+        return db.sync({ force: true });
+    });
+
+    beforeEach('Create app', function () {
+        app = require('../../../server/app')(db);
+        User = db.model('user');
+        Order = db.model('orders');
+    });
+
+	beforeEach('create user', function () {
+		return User.create({
+			first_name: "foo",
+			last_name: "bar",
+			email: "foo@bar.com",
+			password: "foobar"
+
+		})
+		.then(function(_user){
+			user = _user;
+		})
+	});
+
+	describe('All users request', function () {
+
+		var agent;
+
+		beforeEach('create agent', function () {
+			agent = supertest.agent(app);
+		});
+
+		it('GET all', function (done) {
+			agent.get('/api/users')
+				.expect(200)
+		        .end(function (err, res) {
+		          if (err) return done(err);
+		          expect(res.body).to.be.instanceof(Array);
+		          expect(res.body).to.have.length(1);
+		          done();
+		        });
+		});
+
+	});
+
+	describe('Single user request', function () {
+
+		var agent;
+
+		beforeEach('create agent', function () {
+			agent = supertest.agent(app);
+		});
+
+		// COME BACK TO THIS TEST
+		it('GET one', function (done) {
+			agent
+				.get('/api/users/1')
+				.expect(200)
+		        .end(function (err, res) {
+		          if (err) return done(err);
+		          expect(res.body.id).to.equal(user.id);
+		          done();
+		    });
+		});
+
+
+	});
+
+	describe('Update user info', function () {
+
+		var agent;
+
+		beforeEach('create agent', function () {
+			agent = supertest.agent(app);
+		});
+
+		it('checks that updates were correctly applied', function (done) {
+			agent
+				.put('/api/users/1')
+				.send({
+					first_name: "blah"
+				})
+				.expect(200)
+				.end(function(err, res){
+		          if (err) return done(err);
+		          expect(res.body.first_name).to.equal("blah")
```

Comment:
this test confirms that the response's first_name is 'blah' but not whether that change persisted in the database